src/util/os_utils.js

```diff
@@ -1280,14 +1280,52 @@ async function is_vmtools_installed() {
     }
 }
 
-
-async function get_kubernetes_dns_name() {
+async function discover_k8s_services(app = 'noobaa') {
```

Comment:
I think this function creates a strong coupling between the YAML files and k8s internals with our code:
1. using the `app=noobaa` label forces us to not change it in the future
2. using the port names in the yaml file forces us to not change it in the future
3. in general, I think that using kubernetes internals to get configuration data is not a good idea. any change in the API can break an existing noobaa deployment 

I suggest using [configMaps](https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/) to pass configuration data, which are the native kubernetes tool for this job.  
For starter, we can manage the configMap internally in noobaa-server but eventually, I think the right approach is to update this data externally from the server to reduce dependencies.

Reply:
@nb-ohad Please open a tech-debt issue and attach it. Describe that we want to be able to retrieve this information from an outside source and use it.

Reply:
All dependency on k8s in this code path is relegated to one function in os_utils.js (discover_k8s_services) so future changes or additions of  new discovery methods will be relatively easy

Reply:
@nimrod-becker I don't know if this is what we want. You cannot total deprive the system from the knowledge that it is running on k8s (not only in this parts of the codebase but as a whole).
I think that converging all decisions based on env should be grouped into one area of the code base and specialised for each target env/distro and that there is not "evil' in doing so.